core/src/main/scala/org/http4s/multipart/MultipartDecoder.scala

```diff
@@ -0,0 +1,62 @@
+package org.http4s
+package multipart
+
+import scala.util.{Try, Success, Failure}
+import scala.util.control._
+import org.http4s._
+import parser._
+import headers._
+import Http4s._
+import org.http4s.util._
+import scalaz.concurrent._
+import scalaz.concurrent.Task._
+
+import scodec.bits.ByteVector
+import scalaz._
+import Scalaz._
+import scalaz.stream._
+import scalaz.stream.Process._
+import org.parboiled2._
+import org.log4s.getLogger
+
+private[http4s] object MultipartDecoder {
+
+  private[this] val logger = getLogger
+  
+  val decoder: EntityDecoder[Multipart] =
+    EntityDecoder.decodeBy(MediaRange.`multipart/*`) { msg =>
+      def gatherParts = {
+        def go(part: Part): Process1[Headers \/ ByteVector, Part] =
+          receive1Or[Headers \/ ByteVector, Part](emit(part)) {
+            case -\/(headers) =>
+              emit(part) fby go(Part(headers, EmptyBody))
+            case \/-(chunk) =>
+              go(part.copy(body = part.body ++ emit(chunk)))
```

Comment:
I considered this.  I think we might end up accruing all the chunks as we build up a Task definition until we run it to flush it to the temp file.  I might not yet be thinking about it correctly.